src/adapter/src/coord/caught_up.rs

```diff
@@ -228,19 +228,36 @@ impl Coordinator {
 
         let mut all_caught_up = true;
 
-        for id in self.controller.compute.collection_ids(cluster.id)? {
-            if id.is_transient() || exclude_collections.contains(&id) {
-                // These have no relation to dataflows running on previous
-                // deployments.
-                continue;
-            }
+        let storage_frontiers = self
+            .controller
+            .storage
+            .active_ingestions(cluster.id)
+            .iter()
+            .copied()
+            .filter(|id| !id.is_transient() && !exclude_collections.contains(id))
+            .map(|id| {
+                let (_read_frontier, write_frontier) =
+                    self.controller.storage.collection_frontiers(id)?;
+                Ok::<_, anyhow::Error>((id, write_frontier))
+            });
 
-            let write_frontier = self
-                .controller
-                .compute
-                .collection_frontiers(id, Some(cluster.id))?
-                .write_frontier;
+        let compute_frontiers = self
+            .controller
+            .compute
+            .collection_ids(cluster.id)?
+            .filter(|id| !id.is_transient() && !exclude_collections.contains(id))
+            .map(|id| {
+                let write_frontier = self
+                    .controller
+                    .compute
+                    .collection_frontiers(id, Some(cluster.id))?
+                    .write_frontier
+                    .to_owned();
+                Ok((id, write_frontier))
+            });
 
+        for res in itertools::chain(storage_frontiers, compute_frontiers) {
```

Comment:
Nit: I think `storage_frontiers.chain(compute_frontiers)` would work here too.

Reply:
Indeed it would! I like using `itertools::chain` in these situations since (to me) it makes it a smidge clearer as these `storage_frontiers` and `compute_frontiers` have "equal status", vs `storage_frontiers` being the main character and `compute_frontiers` getting bolted on. Maybe that's only how I think about it, though. 😅

Reply:
Fair enough!